esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -418,19 +439,53 @@ def visualize_and_save_weights(weights: 'xr.DataArray', cfg: dict,
     log_provenance(caption, filename_data, cfg, ancestors)
 
 
+def get_variable_groups(contribution: str, cfg: dict) -> list:
+    """Return a list of variable groups with contributions > 0."""
+    if cfg.get(contribution, None) is not None:
+        return [key for key, value in cfg[contribution].items() if value > 0]
+    return []
+
+
 def main(cfg):
     """Perform climwip weighting method."""
     models, observations = read_metadata(cfg)
 
-    variable_groups = models.keys()
+    variable_groups_independence = get_variable_groups(
```

Comment:
So instead of using a list to store the variable groups we want to analyse, we can just use the parsed dict directly. That will save us having to import that again later on. 

```suggestion
    contributions_independence = parse_contributions(
```